https://phabricator.services.mozilla.com/D209108
https://github.com/mozilla/gecko-dev/commit/33cdc6655b0de44cb7a431216dcbb0d5a552aec6

clang 19 will report error if w/o this patch:

  src/third_party/mozjs-60/extract/js/src/threading/ExclusiveData.h:124:33: error: reference to non-static member function must be called

diff --git a/src/third_party/mozjs-60/extract/js/src/threading/ExclusiveData.h b/src/third_party/mozjs-60/extract/js/src/threading/ExclusiveData.h
index 25b977e..379a509 100644
--- a/src/third_party/mozjs-60/extract/js/src/threading/ExclusiveData.h
+++ b/src/third_party/mozjs-60/extract/js/src/threading/ExclusiveData.h
@@ -120,13 +120,6 @@ class ExclusiveData
         release();
     }
 
-    ExclusiveData(ExclusiveData&& rhs)
-      : lock_(mozilla::Move(rhs.lock))
-    {
-        MOZ_ASSERT(&rhs != this, "self-move disallowed!");
-        new (mozilla::KnownNotNull, value_.addr()) T(mozilla::Move(*rhs.value_.addr()));
-    }
-
     ExclusiveData& operator=(ExclusiveData&& rhs) {
         this->~ExclusiveData();
         new (mozilla::KnownNotNull, this) ExclusiveData(mozilla::Move(rhs));
